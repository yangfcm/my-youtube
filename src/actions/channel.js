import { defAxios as axios } from "../settings";
import {
  FETCH_CHANNEL,
  FETCH_CHANNEL_SUBSCRIPTION,
  FETCH_CHANNEL_INTRO,
  SUBSCRIBE_CHANNEL,
  UNSUBSCRIBE_CHANNEL,
  CATCH_ERROR,
} from "./types";
import { DEFAULT_ERROR_MSG } from "./default-error-msg";
