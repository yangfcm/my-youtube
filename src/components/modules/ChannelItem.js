import React from "react";
import { Link } from "react-router-dom";

const ChannelItem = ({ channel }) => {
  return (
    <div className="card">
      <Link to={`/channel/${channel.snippet.resourceId.channelId}`}>
        <div className="pt-3 d-flex justify-content-center">
          <img
            src={channel.snippet.thumbnails.medium.url}
            alt={channel.title}
            className="card-img-top rounded-circle"
            style={{ width: "80%", height: "auto" }}
          />
        </div>
      </Link>
      <div className="card-body">
        <Link to={`/channel/${channel.snippet.resourceId.channelId}`}>
          <h5 className="card-title text-center">{channel.snippet.title}</h5>
        </Link>
      </div>
    </div>
  );
};

export default ChannelItem;
