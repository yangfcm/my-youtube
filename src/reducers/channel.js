import {
  FETCH_CHANNEL,
  FETCH_CHANNEL_SUBSCRIPTION,
  FETCH_CHANNEL_INTRO,
  SUBSCRIBE_CHANNEL,
  UNSUBSCRIBE_CHANNEL,
} from "../actions/types";

export const channelReducer = (state = {}, action) => {
  switch (action.type) {
    case FETCH_CHANNEL:
      return {
        ...state,
        channels: action.payload,
      };
    case FETCH_CHANNEL_INTRO:
      return {
        ...state,
        channelIntro: action.payload,
      };
    case FETCH_CHANNEL_SUBSCRIPTION:
      return {
        ...state,
        isSubscribed: action.payload.isSubscribed,
      };
    case SUBSCRIBE_CHANNEL:
      return {
        ...state,
        subscribedChannel: action.payload,
      };
    case UNSUBSCRIBE_CHANNEL:
      return {
        ...state,
        subscribedChannel: null,
      };
    default:
      return state;
  }
};
