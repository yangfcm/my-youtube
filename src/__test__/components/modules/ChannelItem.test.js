import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";
import ChannelItem from "components/modules/ChannelItem";
import { channelItemData } from "__test__/fixtures/channel";

describe("Test ChannelItem component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ChannelItem channel={channelItemData} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render ChannelItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("links should have correct url", () => {
    expect(wrapper.find(Link).at(0).prop("to")).toBe(
      `/channel/${channelItemData.snippet.resourceId.channelId}`
    );
    expect(wrapper.find(Link).at(1).prop("to")).toBe(
      `/channel/${channelItemData.snippet.resourceId.channelId}`
    );
  });

  it("should render correct channel image", () => {
    expect(wrapper.find("img").prop("src")).toBe(
      channelItemData.snippet.thumbnails.medium.url
    );
  });

  it("should render correct channel title", () => {
    expect(wrapper.find("h5").text()).toContain(channelItemData.snippet.title);
  });
});
