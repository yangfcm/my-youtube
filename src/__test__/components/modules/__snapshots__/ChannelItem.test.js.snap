// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChannelItem component should render ChannelItem component 1`] = `
<div
  className="card"
>
  <Link
    to="/channel/UCPIAvE0MO9pyWGNh5OoYInQ"
  >
    <div
      className="p-3 text-center"
    >
      <img
        className="rounded-circle"
        src="https://yt3.ggpht.com/-CwIKK_-YUE4/AAAAAAAAAAI/AAAAAAAAAAA/qt9fyxUhwBM/s240-c-k-no-mo-rj-c0xffffff/photo.jpg"
        style={
          Object {
            "height": "auto",
            "maxWidth": "100%",
          }
        }
      />
    </div>
  </Link>
  <div
    className="card-body"
  >
    <Link
      to="/channel/UCPIAvE0MO9pyWGNh5OoYInQ"
    >
      <h5
        className="card-title text-center"
      >
        Aussie English
      </h5>
    </Link>
  </div>
</div>
`;
